internal/cephfs/clone.go

```diff
@@ -60,25 +64,28 @@ func createCloneFromSubvolume(ctx context.Context, volID, cloneID volumeID, volO
 			if err = purgeVolume(ctx, cloneID, cr, volOpt, true); err != nil {
 				util.ErrorLog(ctx, "failed to delete volume %s: %v", cloneID, err)
 			}
-			if err = unprotectSnapshot(ctx, parentvolOpt, cr, snapshotID, volID); err != nil {
-				// Incase the snap is already unprotected we get ErrSnapProtectionExist error code
-				// in that case we are safe and we could discard this error and we are good to go
-				// ahead with deletion
-				if !errors.Is(err, ErrSnapProtectionExist) {
-					util.ErrorLog(ctx, "failed to unprotect snapshot %s %v", snapshotID, err)
+			if !autoProtect {
+				if err = unprotectSnapshot(ctx, parentvolOpt, cr, snapshotID, volID); err != nil {
```

Comment:
to keep the code changes minimal, as the `volOptions` are passed to both protect and unprotect function how about checking `checkSubvolumeHasFeature` inside `protect` and `unprotect` function?

Reply:
Checking outside might prevent many extra calls to `protectSnapshot` and `unprotectSnapshot`. Thoughts?

Reply:
Checking inside will check for the feature on subvolume if the feature is present the call will be treated as a no-op

Reply:
Thanks for the input. I'll make the changes.